src/network/event_loop.rs

```diff
@@ -165,12 +114,54 @@ impl EventLoop {
 		&mut self,
 		event: SwarmEvent<
 			BehaviourEvent,
-			EitherError<
-				EitherError<
-					EitherError<EitherError<std::io::Error, std::io::Error>, void::Void>,
-					ping::Failure,
+			itertools::Either<
+				itertools::Either<
+					itertools::Either<
+						itertools::Either<
+							itertools::Either<
+								itertools::Either<
+									itertools::Either<
+										itertools::Either<
+											ConnectionHandlerUpgrErr<
+												itertools::Either<
+													InboundHopFatalUpgradeError,
```

Comment:
wow :)

Reply:
Maybe we can replace qualified path with use for `Either` and hide complexity a little :S

Reply:
it's getting more and more beautiful

Reply:
also it seems that handle_event is not async and we can remove it